Automattic/jetpack
# Allow is_truthy to properly handle occasional objects.

Fixes a fatal error that can occur on WordPress.com API requests when third party plugins add data to responses. 

## Proposed changes:
* Add an additional check to is_truthy for non-object arguments.

### Other information:

## Jetpack product discussion


p1703178139555289-slack-C02ECE3ML

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






To test this change you will need to sandbox public-api.wordpress.com, update wpcom with this PR, as well as introduce a new filter into your test Jetpack site:

```php
add_action( 'wpcom_api_site_roles', 'add_bogus_stdclass_roles' );
function add_bogus_stdclass_roles( $roles ) {
    $bogus_role = new WP_Role( 'bogus_role', 'Bogus role' );
    $bogus_role->capabilities = new stdClass();

    $bogus_role->capabilities->pmpro_role_1 = new stdClass();
    $bogus_role->capabilities->pmpro_role_1->read = 1 ;

    $roles[] = $bogus_role;
    return $roles;
}
```

After adding this filter and updating wpcom, use the WordPress.com Developer console to query the site's API using the roles endpoint `v1.1/sites/SITE_ID/roles`. Observe no fatals in the sandbox. Without this patch but with the filter installed, you will see a fatal on your sandbox.
